# CSE110 Lab Assignment 3

This Assignment is to help you develop your concept of Strings in Python.

In [5]:
string_1="abcdefg"
number=4
x=''
for i in range(number):
    x+=string_1[i]
z=''
for i in range(len(x)-1,-1,-1):
    z+=x[i]
y=z+string_1[number:]
print(y)


dcbaefg








##Write your name, student ID and CSE110 theory section number below:







In [ ]:
#STUDENT NAME:
#STUDENT ID:
#CSE110 THEORY SECTION: 

In [24]:
user_input=6 #5 ===4
for line in range(1,user_input):
    for col in range(1,line+1):
        print(line,end="")
        print(col,end="")
    print()                 

11
2122
313233
41424344
5152535455


## Write the code in Python to do the following tasks:

**<font color='red'>[MUST MAINTAIN VARIABLE NAMING CONVENTIONS FOR ALL THE TASKS]</font>**


### Task 1 

Write a Python program that will ask the user to input a string (containing exactly one word). Then your program should print subsequent substrings of the given string as shown in the examples below.

=====================================================================

**Example 1:**\
Input: BANGLA\
Output:\
B\
BA\
BAN\
BANG\
BANGL\
BANGLA

=====================================================================

**Example 2:**\
Input: DREAM\
Output:\
D\
DR\
DRE\
DREA\
DREAM

=====================================================================

**Hints(1):** Need to use "for loop" for this task.

**Hints(2):** Need to use print() function for printing newlines.

For example:\
print(1)\
print(2)

Output:\
1\
2

=====================================================================

We need use print(end = "") to skip printing the additional newline.

For example:\
print(1, end =" ")\
print(2)

Output:(prints the following output right next to the previous one)\
12

=====================================================================






In [1]:
#to do
s=input("Enter the number: ")
for i in range(len(s)):
    print(s[0:i+1])


Enter the number: BANGLA
B
BA
BAN
BANG
BANGL
BANGLA


### Task 2

Write a Python program that will ask the user to enter a  word as an input. 

* If the length of the input string is less than 4, then your program should print the same string as an output. 
* If the input string’s length is greater than 3, then your program should add "er" at the end of the input string. 
* If the input string already ends with "er", then add "est" instead. 
* If the input string already ends with "est", then your program should print the same input string as an output.

=====================================================================

**Example 1:**\
Input: strong\
Output: stronger

=====================================================================

**Example 2:**\
Input: stronger\
Output: strongest

=====================================================================

**Example 3:**\
Input: strongest\
Output: strongest

=====================================================================

**Example 4:**\
Input: abc\
Output: abc

=====================================================================

In [2]:
#to do
var1=input("Enter the string:")
p=len(var1)

if(p<4):
    print(var1)
elif(p>3):
    if(var1.endswith("est")):
        print(var1)
    elif (var1.endswith("er")):
        print(var1.replace("er","est"))
    elif(not(var1.endswith("er"))):
        print(var1+'er')


Enter the string: stronger
strongest


### Task 3

Write a Python program that will ask the user to input a string (containing exactly one word). Then print the ASCII code for each character in the String using the ord() function. 

To check if your program is working correctly or not, you can find a list of all correct values from the following website. Look at “Dec” and “Char” columns only, ignore other columns.\
link: http://www.asciitable.com/

=====================================================================

**Example 1:**\
Input: Programming\
Output:\
P : 80\
r : 114\
o : 111\
g : 103\
r : 114\
a : 97\
m : 109\
m : 109\
i : 105\
n : 110\
g : 103

=====================================================================

**Example 2:**\
Input: hunger\
Output:\
h : 104\
u : 117\
n : 110\
g : 103\
e : 101\
r : 114 

=====================================================================

In [4]:
#to do
p=input("Enter the string: ")
for i in range(len(p)):
    print(p[i],":",ord(p[i]))
    

Enter the string: Programming
P : 80
r : 114
o : 111
g : 103
r : 114
a : 97
m : 109
m : 109
i : 105
n : 110
g : 103


### Task 4

Take a string as a input from the user with all small letters. Then print the next alphabet in sequence for each alphabet found in the input.

=====================================================================


*Hint: You need to use functions ord() and chr(). The ASCII value of ‘a’ is 97 and ‘z’ is 122.*

=====================================================================

**Example 1:**\
Input: abcd\
Output: bcde

=====================================================================

**Example 2:**\
Input: the cow\
Output: uif!dpx

=====================================================================

**Example 3: <font color='red'>[Must fulfil this criteria]</font>**\
Input: xyzabc\
Output: yzabcd

=====================================================================


In [12]:
s1=input("enter the string: ")

for i in range(len(s1)):
    value=ord(s1[i])
    value=value+1
    if value>122:
        value=97
    print(chr(value),end="")

enter the string: xyzabc
yzabcd

### Task 5

Write a Python program that will ask the user to enter two strings (s1, s2) as an input. Then create a mixed string with alternative characters from each string. Any leftover chars will be appended at the end of the resulting string.

=====================================================================

*Hint: For adding the leftover characters you may use string slicing.*

=====================================================================

**Example 1:**\
Input:\
"ABCD"\
"efgh"

Output:\
AeBfCgDh

=====================================================================

**Example 2:**\
Input:\
"ABCDENDFGH"\
"ijkl"

Output:\
AiBjCkDlENDFGH

=====================================================================

**Example 3:**\
Input:\
"ijkl"\
"ABCDENDFGH"

Output:\
iAjBkClDENDFGH

=====================================================================


In [2]:
#todo
s1="ABCDENDFGH"
s2="ijkl"

s=""
small=""
large=""
if (len(s1)>len(s2)):
    small=s2
    large=s1
else:
    small=s1
    large=s2
for i in range(len(small)):
    s=s+s1[i]+s2[i]
s=s+large[i+1:len(s1)]
print(s)


AiBjCkDlENDFGH


### Task 6
Given a string, create a new string with all the consecutive duplicates **removed**. 

=====================================================================

**Hint:** You may make a new string to store the result. You can check whether the current character and the next character are the same, then add that character to the new string.

=====================================================================


**Sample Input**:\
AAABBBBCDDBBECE

**Sample Output:**\
ABCDBECE

=====================================================================


In [5]:
#to do
s=input("Enter the string: ")
s2=s[0]
for i in range(1, len(s)):
    if (s[i]!=s[i-1]):
        s2=s2+s[i]
print(s2)

Enter the string: AAABBBBCDDBBECE
ABCDBECE


In [13]:
#7 number practice 
string_1="72418"
n=4
x=''
for i in range(n,0,-1):
    x+=string_1[i]
for i in string_1:
    if i not in x:
        x+=i
print(x)

81427


### Task 7

Write a python program that takes 2 inputs from the user, where the first input is a string with length greater than 1. The second input is the index of the first given string from where you have to start reversing. Then print the new string back to the user.

=====================================================================

**Sample Input 1:**

72418

4

**Sample Output 1:**

81427


**Explanation**: Our second input, index '4' is the last index of our first input String '72418', hence the entire string is reversed giving us '81427'.

=====================================================================

**Sample Input 2:**

12345 

2

**Sample Output 2:**

32145

**Explanation**: The second input is '2' so we have to reverse from index 2 of our first input. The 2nd index of our first input String is '3', index 1 is '2' and index 0 is '1'. Hence, if we reverse indexes 0 to 2, we get '321'. Index 3 and 4 which is '4' and '5' respectively remains unchanged hence our final output is '32145'.

=====================================================================

**Sample Input 3:**

aBcd1234defg

5

**Sample Output 3:**

21dcBa34defg

**Explanation**: From our first input String 'aBcd1234defg', 

index 0 = 'a'

index 1 = 'B'

index 2 = 'c'

index 3 = 'd'

index 4 = '1'

index 5 = '2'

index 6 = '3'

Since our second input is 5, index 0 to index 5 is reversed and we have '21dcBa' and the rest is unchanged from indexes 6 to 11 ('34defg'). Therefore, we have '21dcBa34defg' finally.

=====================================================================


In [7]:
#to do
s1=input("Enter the string: ")
n=int(input("Enter the number: "))
s2=" "
for i in range (n,-1,-1):
    s2=s2+s1[i]
if(n<len(s1)):
    s2=s2+s1[n+1:len(s1)]
print(s2)

Enter the string: aBcd1234defg
Enter the number: 5
 21dcBa34defg


### Task 8

Write a python program that splits a given string on a given split character. The first input is a String and the second input is the character that will be used to split the first String. <font color='red'>[You cannot use the built-in split function]</font>

=====================================================================

**Sample Input 1:**<br/>
This-is-CSE110<br/>
-<br/>

**Sample Output 1:**<br/>
This<br/>
is<br/>
CSE110

**Explanation**: The second input which is the character '-', is used to split or divide the first input String 'This-is-CSE110' into 'This', 'is' and 'CSE110' which are printed individually in seperate lines.

=====================================================================

**Sample Input 2**:
tom@gmail,harry@yahoo,bob@gmail,mary@gmail

,

**Sample Output 2**:

tom@gmail

harry@yahoo

bob@gmail

mary@gmail

=====================================================================

In [2]:
s=input("enter the string")
char=("-")
wrd=""
for i in range(len(s)):
    if ( char==s[i]):
        print(wrd)
        wrd=""
    else:
        wrd=wrd+s[i]
print(wrd)


enter the stringThis-is-CSE110
This
is
CSE110


### Task 9

Suppose you are given two strings, s1, and s2. Now, print a new string made up of the last characters and then the first characters of the input strings.

=====================================================================

**Sample Input 1**:

s1 = new

s2= string

**Sample Output 1**:

gwsn

**Explanation:** The last character of the String s2 is 'g'. The last character of the String s1 is 'w'. The first character of the String s2 is 's'. The first character of the String s1 is 'n'. Together they give us the ouput we want 'gwsn'.

=====================================================================

**Sample Input 2**:

s1 = abcd

s2= efgh

**Sample Output 2**: 

hdea

**Explanation:** The last characters of the Strings s2 and s1 is 'h' and 'd' respectively while the first characters of the Strings is 'e' and 'a' respectively. Together they give us the ouput we want 'gwsn'.

=====================================================================


In [6]:
#to do
s1=input("Enter string: ")
s2=input("Enter string: ")
s=''
s=s2[-1]+s1[-1]+s2[0]+s1[0]
print(s)

Enter string: abcd
Enter string: efgh
hdea


### Task 10

Write a Python program that takes a String as input from the user, removes the characters at even index and prints the resulting String in uppercase.

=====================================================================

**Sample Input 1**:

String

**Sample Output 1**: 

TIG

**Explanation**: The characters 'S', 'r' and 'n' are at index positions 0, 2, and 4 respectively. Hence they are removed and the remaining characters 'tig' are capitalized giving us output 'TIG'.

=====================================================================

**Sample Input 2**:

abcd

**Sample Output 2**: 

BD

=====================================================================


In [16]:
#to do
string=input("Enter the string:")
newstring=""
for i in range (len(string)):
    if(i%2!=0):
        newstring=newstring+string[i].upper()
print(newstring)

Enter the string: string
 TIG


### Task 11

Write a python program that takes two inputs. The first input is a string and the second input is a number. If the number is even then concatenate the given string two times the given number and if the number is odd then concatenate the given string three times the given number.

=====================================================================

**Sample Input 1**:

CSE110

4

**Sample output 1**:

CSE110CSE110CSE110CSE110CSE110CSE110CSE110CSE110

**Explanation**: The second input which is the number 4 is even, therefore the first string input 'CSE110' is concatenated(joined together) 4*2 = 8 times.

=====================================================================

**Sample Input 2**:

CSE110

3

**Sample Output 2**:

CSE110CSE110CSE110CSE110CSE110CSE110CSE110CSE110CSE110

=====================================================================


In [ ]:
#to do
string=input("Enter the string: ")
n=int(input("Enter the number: "))
times=0
if(n%2==0):
    times=2
else:
    times=3
print(string*n*times)

### Task 12

Write a python program that takes 2 inputs from the user. The first input is a string and the second input is a letter. The program should remove all occurences of the letter from the given string and print the output. If the letter is not found in the string and the length of string is greater than 3, then remove the first letter and last letter of the given string and print it. Otherwise print the string as it is. [You can assume that all the input will be in lowercase letter]

=====================================================================


**Sample Input 1**:

tanjiro kamado

a

**Sample output 1**:

tnjiro kmdo

**Explanation**: All 3 instances of the character 'a' is removed from the input String 'tanjiro kamado' to give us output 'tnjiro kmdo'.

=====================================================================

**Sample Input 2**:

eren yeager

k


**Sample Output 2**:

ren yeage

**Explanation**: The character 'k' is absent in the first input String 'eren yeager' and it's length is 11 which is greater than 3 therefore the first character 'e' and the last character 'r' is removed. Hence, the final String is 'ren yeage'.

=====================================================================

Sample Input 3:

hi

a

Sample Output 3:

hi

**Explanation**: The letter 'a' is not found in our first input 'hi', the length of which is 2. Since the character is not present and the length is less than 3, we print the String 'hi' as it is.

=====================================================================


In [4]:
#to do
s=input("Enter the string: ")
char=input("Enter char: ")
s2=""
for i in range(len(s)):
    if(s[i]!=char):
        s2=s2+s[i]
print(s2)


Enter the string: python
Enter char: y
pthon


## Optional Tasks (13 - 15) [Ungraded]

### Task 13


Write a python program that takes a string as an input from the user and then modifies the string in such a way that the string always starts with an uppercase letter and the case of each subsequent letter is the opposite of the previous letter (uppercase character followed by a lowercase character followed by an uppercase character and so on).  Finally the modified string is printed to show the user.

*Hints:  Flags/counters can be used to manage uppercase-lowercase.*

=====================================================================

**Sample Input 1:**\
Python programming is very easy

**Sample Output 1:**\
PyThOn PrOgRaMmInG iS vErY eAsY

=====================================================================

**Sample Input 2:**\
I&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     love       &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Python &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Programming

**Sample Output 2:**\
I&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     lOvE       &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;pYtHoN&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;pRoGrAmMiNg

=====================================================================

**Sample Input 3:**\
CSE110 Course

**Sample Output 3**\
CsE110 cOuRsE



=====================================================================

**Sample Input 4:**\
c

**Sample Output 4:**\
C

=====================================================================


In [1]:
user_input=("Python programming ").upper()
# user_input=user_input.upper()
ans=""
counter=1
for char in user_input:
    if char>="A" and char<="Z":
        print(char,"Alphabate")
        if (counter%2==0):
            ans=ans+char.lower()
            print(ans)
        else:
            ans=ans+char
            print(ans)
        counter+=1    
    else:
        print(char,"space")
        ans=ans+char


P Alphabate
P
Y Alphabate
Py
T Alphabate
PyT
H Alphabate
PyTh
O Alphabate
PyThO
N Alphabate
PyThOn
  space
P Alphabate
PyThOn P
R Alphabate
PyThOn Pr
O Alphabate
PyThOn PrO
G Alphabate
PyThOn PrOg
R Alphabate
PyThOn PrOgR
A Alphabate
PyThOn PrOgRa
M Alphabate
PyThOn PrOgRaM
M Alphabate
PyThOn PrOgRaMm
I Alphabate
PyThOn PrOgRaMmI
N Alphabate
PyThOn PrOgRaMmIn
G Alphabate
PyThOn PrOgRaMmInG
  space


### Task 14 
An anagram is a play on words created by rearranging the letters of the original word to make a new word or phrase.
So we can say two words are anagrams if they contain all of the same letters, but in a different order.


Write a python program that takes two strings from the user and tells if they are anagram or not.

=====================================================================

<b>Sample Input 1: </b><br/>
dusty<br/>
study<br/>
<b>Sample Output 1: </b> <br/>
They are anagram<br/>

=====================================================================

<b>Sample Input 2: </b><br/>
dustyyy<br/>
study<br/>
<b>Sample Output 2: </b><br/>
They are not anagram

In [8]:
string1="dustyy"
string2="study"
# str=""
flag=True

for i in string1:
    counter1=0
    counter2=0
    if len(string1)==len(string2):
        for i in string1:
            counter1+=1
        for i in string2:
            counter2+=1
            
        if ( counter1!=counter2):
            flag=False
    else:
        flag=False
if flag:
    print("anagram")
else:
    print("Not anagram")


Not anagram


### Task 15

Write a python program thats prints the largest consecutive sequence from a given string S. For example, if S = ‘AABBBCCAAAA’, print ‘AAAA’. If more than one largest consecutive sequence exists, print the first one. Print NONE if no consecutive sequence is found.

=====================================================================

**Sample Input 1:**<br/>
AABBBCCAAAA<br/>
**Sample Output 1:**<br/>
AAAA<br/>

=====================================================================

**Sample Input 2:**<br/>
AABBCC<br/>
**Sample Output 2:**<br/>
AA<br/>

=====================================================================

**Sample Input 3:**<br/>
ABCD<br/>
**Sample Output 3:**<br/>
NONE<br/>

=====================================================================

In [1]:
#to do
str = input("Enter the letter : ")
flag =False
max = ''
temp = str[0]
for i in range(1,len(str)):
    if str[i-1] == str[i]:
        flag=True
        temp+=str[i]
        
        if len(max)<len(temp):
            max=temp
    else:
        temp = str[i]
if flag==True:
    print(max)
else:
    print("None")


Enter the letter : AABBBBBBAAAA
BBBBBB
